src/domain/opendatafabric/src/identity/dataset_refs.rs

```diff
@@ -740,3 +741,140 @@ impl std::cmp::PartialOrd for DatasetRefAny {
         Some(self.cmp(other))
     }
 }
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum DatasetRefPattern {
+    Ref(DatasetRef),
+    Pattern(Option<AccountName>, DatasetNamePattern),
+}
+
+impl DatasetRefPattern {
+    pub fn is_match(dataset_ref: DatasetRef, pattern: &str) -> Result<bool, InvalidPatternError> {
+        Like::<false>::like(dataset_ref.to_string().as_str(), pattern)
+    }
```

Comment:
1) I still don't see any tests for this function

2) `dataset_ref: DatasetRef` can be passed by reference instead of by value

3) Why is this a static function and not a method? Wouldn't it be better to have it as:
  `fn is_match(&self, dataset_ref: &DatasetRef)`

    Otherwise the user of this API will be forced to extract `pattern: &str` somehow from `DatasetRefPattern`, which will involve complex matching / destructuring etc.

4) Consider what will happen if `dataset_ref` is in form of `did:odf:xxx`. In RFC we said we don't ever want to pattern-match IDs, but there's nothing protecting us against this here. (need tests!)

Reply:
5. Since we already validated the pattern via our strict grammar in theory it should never return `InvalidPatternError`